tools/serverpod_cli/bin/util/yaml_docs.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:source_span/source_span.dart';
+
+class YamlDocumentationExtractor {
+  final List<String> lines;
+
+  YamlDocumentationExtractor(String yaml) : lines = yaml.split('\n');
+
+  List<String>? getDocumentation(SourceLocation keyStart) {
+    var documentation = <String>[];
+    var docStartExp = RegExp('^${''.padLeft(keyStart.column)}###(.*)');
+    for (var i = keyStart.line - 1; i >= 0; i--) {
```

Comment:
I may be missing something here. But will this not continue the search upwards past non-documented fields?

E.g. will the comment for fieldA be matched for fieldB?
```yaml
class: Foo
fields:
  ### Is this matched?
  fieldA: String
  fieldB: String
```

Reply:
The algorithm executes line for line, starting with the line above the key.
 
`else if (line.isNotEmpty && RegExp(r'^\ *?[^#]*$').hasMatch(line))` a few lines down would break the loop when searching the docs for fieldB. So fieldB would not have a documentation.

Reply:
So hard to read those regexps.

Reply:
I'm always using [this](https://regex101.com/r/CVky9y/1) side.